src/MultiFormats/MultiAddress.cs

```diff
@@ -0,0 +1,515 @@
+        /// </param>
+        public MultiAddress(string s) : this()
+        {
+            if (string.IsNullOrWhiteSpace(s))
+                return;
+
+            Read(new StringReader(s));
+        }
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="MultiAddress"/> class from the
+        ///   specified <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   A <see cref="Stream"/> containing the binary representation of a
+        ///   <b>MultiAddress</b>.
+        /// </param>
+        /// <remarks>
+        ///   Reads the binary representation of <see cref="MultiAddress"/> from the <paramref name="stream"/>.
+        ///   <para>
+        ///   The binary representation is a sequence of <see cref="NetworkProtocol">network protocols</see>.
+        ///   </para>
+        /// </remarks>
+        public MultiAddress(Stream stream)
+            : this()
+        {
+            Read(stream);
+        }
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="MultiAddress"/> class from the
+        ///   specified <see cref="IPAddress"/>.
+        /// </summary>
+        public MultiAddress(IPAddress ip)
+            : this()
+        {
+            var type = ip.AddressFamily == AddressFamily.InterNetwork
+                ? "ip4"
+                : "ip6";
+            Read(new StringReader($"/{type}/{ip}"));
+        }
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="MultiAddress"/> class from the
+        ///   specified <see cref="IPEndPoint"/>.
+        /// </summary>
+        public MultiAddress(IPEndPoint endpoint)
+            : this()
+        {
+            var type = endpoint.AddressFamily == AddressFamily.InterNetwork
+                ? "ip4"
+                : "ip6";
+            Read(new StringReader($"/{type}/{endpoint.Address}/tcp/{endpoint.Port}"));
+        }
+
+        /// <summary>
+        ///   Creates a deep copy of the multi address.
+        /// </summary>
+        /// <returns>
+        ///   A new deep copy.
+        /// </returns>
+        public MultiAddress Clone() { return new MultiAddress(ToString()); }
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="MultiAddress"/> class from the
+        ///   specified byte array.
+        /// </summary>
+        /// <param name="buffer">(
+        ///   A byte array containing the binary representation of a
+        ///   <b>MultiAddress</b>.
+        /// </param>
+        /// <remarks>
+        ///   Reads the binary representation of <see cref="MultiAddress"/> from the <paramref name="buffer"/>.
+        ///   <para>
+        ///   The binary representation is a sequence of <see cref="NetworkProtocol">network protocols</see>.
+        ///   </para>
+        /// </remarks>
+        public MultiAddress(byte[] buffer)
+            : this()
+        {
+            if (buffer == null || buffer.Length == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)